### ISSUE TITLE
Game crash with boiler multiblock

### ISSUE THREAD

**Issue by House239** [PIPPIP5789/Revolution-Of-Steam#2]
So i tried to input water into a boiler with immersive engineering pipes but when i connected the pipes to the boiler water input the game crashed. The game worked just fine when the pipes were empty, but as soon as the pipes had water in them, the game crashed.
[crash-2024-10-24_18.40.49-server.txt](https://github.com/user-attachments/files/17509563/crash-2024-10-24_18.40.49-server.txt)

**Comment by PIPPIP5789** [comment]
First of all, thanks for the report!

This is an odd error, to say the least, given that the Boiler Water Input's Capability (the thing that crashed the game) is, in fact, implementing the correct interface. I'll run some tests on my end to see if I can replicate the error. Was there anything else connected to the pipes that might've caused an issue?

**Comment by House239** [comment]
I had a pump from immersive engineering that pumped out water from a nearby lake and brought it to the boiler with the pipes. The game only crashed when the pump was active and pumping out water.

**Comment by House239** [comment]
Ok this is weird, i made another modpack with way less mods but with the most important ones still there (because it takes way too long to get to the main modpack) and the bug still happened. I didn't even fully connect the pipes to the boiler and the game crashed. It just happened when i turned on the pump.
[crash-2024-10-25_10.29.12-server.txt](https://github.com/user-attachments/files/17517849/crash-2024-10-25_10.29.12-server.txt)
I'll do some more testing to figure this out.

**Comment by House239** [comment]
I found what caused the issue: Immersive Technology. That mod did something to change the IE pipes that made it incompatible with Revolution of Steam. (Immersive Technology is an IE addon mod). The setup worked just fine when i got rid of that mod. This is a problem, because i would really like to keep Immersive Technology in my modpack, but i also want to keep Revolution of Steam in the modpack. I think what i probably have to do is to set up a warning in the questbook that IE pipes crash the game, since there are some multiblocks that use the IE pipes and the IE pump.
![2024-10-25_10 41 43](https://github.com/user-attachments/assets/73824399-8cfa-4523-8434-d2644e069369)
Here's the setup I used. It only crashes the game when Immersive Technology is present.

**Comment by PIPPIP5789** [comment]
Yep, that seems to line up with the crash report. With that said, that doesn't explain why it's happening... I'm going to continue to look into this; I can't promise an update fixing it anytime soon, just because of all the reworking going on under the hood to try and get some of the other multiblocks like the Vat working, but hopefully I'll have found a fix for it before the next update, whenever that ends up being.

**Comment by House239** [comment]
So i just noticed that there's a config option in Immersive Technology that disables its replacement for the IE pipes that caused a crash with ROS.
![config](https://github.com/user-attachments/assets/d05c348e-5ee1-4be0-a473-86f4186934cb)
This is the config option. If it's set to true, the game crashes when using IE pipes with the criteria that i have previously mentioned.

Now that i found a temporary solution for this bug there's only a few more problems such as JEI support not working that stop this mod from being in every one of my 1.12 modpacks. (also i should probably open an issue about the JEI support now that i think about it since the source code still has code for JEI support)

**Comment by PIPPIP5789** [comment]
I'm still working on trying to fix it, so hopefully it'll be better by the next update... though that also might be a while, so I'm glad you found a fix.
